### PR TITLE
CI: Create nightly AppImages

### PR DESCRIPTION
**Pull request by cg2121**
### Description
This is a test to see if they work.

### Motivation and Context
Revival of https://github.com/obsproject/obs-studio/pull/1926

### How Has This Been Tested?
Downloaded and ran AppImage.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### TODO
- [ ] Get VAAPI working.

### CODE REVIEW

**Comment by cg2121** [comment]
Everything is working fine, except VAAPI, which is still a problem.

File: CI/install/AppDir/AppRun [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# libobs.so.0 loads its resources relative to the user's current working directory (cwd).
+# Hence we need to cd to usr/ inside the AppDir before executing the application.
+# A proper fix would be to make libobs.so.0 load its resources from a path relative to itself.
+# But seemingly OBS on the Mac works the same, so we can live with this for now.
```
Comment (DDRBoxman) [inline CI/install/AppDir/AppRun]: OSX now points to the paths in the bundle in various places, but yeah that was the old behavior until catalina broke it
https://github.com/obsproject/obs-studio/pull/2125/files#diff-5284387c0b6ea42e30931b42c288edecR39

**Comment by derrod** [comment]
Maybe not really an issue, but how do plugins work with AppImages?

Are they like mounted file systems you could write to?

**Comment by kkartaltepe** [comment]
> Maybe not really an issue, but how do plugins work with AppImages?
> 
> Are they like mounted file systems you could write to?

Please refer to the original PR https://github.com/obsproject/obs-studio/pull/2441#issuecomment-593142906

**Comment by probonopd** [comment]
Hi, great job, I'd like to test. Still new to this Azure Pipelines thing, where can the AppImage be downloaded for testing?

**Comment by dodgepong** [comment]
I believe the latest build is here: https://dev.azure.com/obsjim/obsjim/_build/results?buildId=2828&view=artifacts&type=publishedArtifacts

Click the down arrow to the right of the "linux_build" line.

**Comment by probonopd** [comment]
Wow, I had not seen that arrow to the right... clicked the text, nothing had happened :-/ thanks!

**Comment by probonopd** [comment]
The download link no longer seems to work.

File: CI/install/AppDir/AppRun (hunk 1 above)
Comment (azubieta) [reply CI/install/AppDir/AppRun:6]: the environment variable `OBS_DATA_PATH` can be used to point the resources path.

**Comment by azubieta** [comment]
I've created a recipe using appimage-builder here https://www.opencode.net/azubieta/obs-appimage/
I'm not familiar with azure pipelines that's why I'm not sending a direct PR.

But I guess you can easily replicated.

**Comment by Fenrirthviti** [comment]
Can you explain why you redid the work from this PR? What is the advantage to your version over the version provided in this PR? If it's fixing issues, can you explain them and how you fixed them so that this PR can be updated? It seems unnecessary to have two appimage implementations.

**Comment by azubieta** [comment]
>> Can you explain why you redid the work from this PR?

Yes, I'm using a new tool for building the AppImage, which makes the whole recipe IMHO more simple and easy to maintain. Also, it allows building the AppImage in a more recent system keeping the backward compatibility.

>> It seems unnecessary to have two appimage implementations.

Agreed

>>  If it's fixing issues, can you explain them and how you fixed them so that this PR can be updated?

The approach that I used only reduces the script complexity and increases the bundle portability. I provide it for illustrative purposes.

I would like it to be checked by the PR author @cg2121

**Comment by DDRBoxman** [comment]
So the difference is:
https://github.com/probonopd/go-appimage
vs.
https://www.opencode.net/azubieta/appimage-builder

I vote we stick with the tool from the guy who came up with appimage

**Comment by kkartaltepe** [comment]
The appimage-builder repository is also dependent on our ppa packaging.

**Comment by azubieta** [comment]
I just updated the AppImage recipe for building the application from sources.

In that repo I'm patching `libobs/obs-nix.c` to fix some portability issues related to libobs. This patch is also useful for the recipe purposed in this PR.


Regarding which tool should you use: appimage-builder is a quite new tool (was released a month ago) but it provides a solution for building AppImages in newer systems keeping the backward compatibility. If you're going to use other tools you should build in an old system for it to run on other old systems. It tries to ease the creation process by having a well-structured recipe. It also supports building AppImage for an architecture different than the one used in the system.

But you should always choose the tool that fits best your needs.

By the way, I'm also part of the AppImage project :)

**Comment by cg2121** [comment]
Since we updated the CI to 18.04, building the AppImage doesn't work anymore. So if the newer solution works with newer distros, that would be great.

**Comment by cg2121** [comment]
@azubieta could your make a PR for your solution, since this doesn't work with our CI anymore? If it works and is simpler, we should definitely use it instead.

**Comment by azubieta** [comment]
@cg2121 I'm not familiar with github actions, but I guess that it's something inevitable to learn about it. Ill take a look.

**Comment by probonopd** [comment]
Just a heads up, I tested @azubieta's AppImage on a Xubuntu 18.04.4 LTS system with Nvidia Driver version 390.132, and I was able to use nvenc right out of the box, which I think is quite an achievement.

**Pull request closed without merging**